gradle/libs.versions.toml

```diff
@@ -4,6 +4,7 @@ jib = "3.4.5"
 spotless = "7.0.3"
 junit = "5.12.2"
 autoservice = "1.1.1"
+log4j2 = "2.24.3"
```

Comment:
do we really want to start explicitly depending on log4j? Any reason not to use j.u.l?

Reply:
JUL static initialization is something that tends to create issues (for example with Wildfly/Jboss), so we'd better avoid it.

Also log4j provides ability to change log level at runtime very easily, so I thought it would be simpler.

Reply:
Yes, I saw the agent itself uses j.u.l and works around the initialization issues by having it's own Log manager. If we tried to switch the upstream agent from using simple (which can't change dynamically) to another logger, then this would work upstream too. But it may be too much work

Reply:
I agree, for this I took inspiration from MS ApplicationInsights that delegates to logback instead of log4j. We could maybe challenge this again upstream and suggest to use either slf4j or an internal logging API in the instrumentation to avoid having to deal with JUL at all in the agent, but that would likely be too much work for little benefit.